client/my-sites/earn/main.jsx

```diff
@@ -68,45 +58,49 @@ class EarningsMain extends Component {
 		}
 
 		return tabs;
-	}
+	};
 
-	getComponent( section ) {
-		switch ( section ) {
+	const getSelectedText = () => {
+		const selected = find( getFilters(), { path: path } );
+		if ( selected ) {
+			return selected.title;
+		}
+
+		return '';
+	};
+
+	const getComponent = ( currentSection ) => {
+		switch ( currentSection ) {
 			case 'ads-earnings':
 				return (
-					<AdsWrapper section={ this.props.section }>
-						<WordAdsHighlightsSection siteId={ this.props.siteId } />
-						<WordAdsEarnings site={ this.props.site } />
+					<AdsWrapper section={ section }>
+						<WordAdsHighlightsSection siteId={ site?.ID } />
+						<WordAdsEarnings site={ site } />
 					</AdsWrapper>
 				);
 			case 'ads-payments':
 				return (
-					<AdsWrapper section={ this.props.section }>
-						<WordAdsPayments site={ this.props.site } />
+					<AdsWrapper section={ section }>
+						<WordAdsPayments site={ site } />
 					</AdsWrapper>
 				);
 			case 'ads-settings':
 				return (
-					<AdsWrapper section={ this.props.section }>
+					<AdsWrapper section={ section }>
 						<AdsSettings />
 					</AdsWrapper>
 				);
 			case 'payments':
-				return <MembershipsSection section={ this.props.section } query={ this.props.query } />;
+				return <MembershipsSection section={ section } query={ query } />;
 			case 'payments-plans':
-				return <MembershipsProductsSection section={ this.props.section } />;
+				return <MembershipsProductsSection section={ section } />;
 
 			case 'refer-a-friend':
 				return <ReferAFriendSection />;
 
 			default:
 				return <Home />;
 		}
-	}
-
-	handleDismissWordAdsError = () => {
-		const { siteId } = this.props;
-		this.props.dismissWordAdsError( siteId );
```

Comment:
This method no longer appears to be used in this component, and so I've removed it.